### DISCUSSION TITLE
급함)타입스크립트 함수를 UI 버튼에서 불러오기

### DISCUSSION THREAD

**Discussion by smartbh** [naverz/zepeto-studio-kor#46] [5. Q&A]
안녕하세요 UI 버튼을 각 오브젝트에 넣어서

상호 작용을 할수 있게 하고싶습니다.

![image](https://user-images.githubusercontent.com/43138813/145446556-cebbf10f-e61f-4638-84f9-359be9f8390d.png)

이렇게 앉을 수 있는 오브젝트에 갔을시 버튼이 뜨게 하는데 까진 성공했습니다.

그러나 버튼에 제페토 스크립트를 주어

버튼이 잘 작동되는지 함수를 짜서 확인 해본결과

![image](https://user-images.githubusercontent.com/43138813/145446629-f3826ba9-0932-42b4-b269-d56ff3a12825.png)
(전체 스크립트)

![image](https://user-images.githubusercontent.com/43138813/145446672-a686eaa9-d861-4edc-8158-7e9edbe1e0f6.png)
버튼의 Function 목록에 함수가 뜨질 않아

해당 기능을 배정해줄수가 없어 질문을 드립니다.

혹시 접근 지정자가 문제여서 그런가 싶어서 public을 주었는데도

목록에서 찾을수가 없어서요

해결하고싶은데 부탁드립니다.

2. 해당 버튼을 눌렀을시

동적으로 생성되는 플레이어 캐릭터의 애니메이터에서

특정 애니메이션 클립 동작을 하게 하고싶습니다.

그런데 아무리 생각해봐도 스크립트에서 캐릭터 애니메이션 컨트롤러에

접근하는 방법이 생각이 안나는데

좋은 예제 혹시 가능할까요?

게임내에서 제공하고 있는 비슷한 기능을 가진

![image](https://user-images.githubusercontent.com/43138813/145447351-c4676b7f-f303-4b57-b49d-9ae15d53236b.png)

![image](https://user-images.githubusercontent.com/43138813/145447389-9514bae2-c9c1-412d-8233-801c3e3ae3c5.png)

기본 UI의 코드를 보려 해도 dll 파일이라 볼 수가 없어서 예제를 찾기 어렵습니다

또한 버튼을 누를때

작게 만들어서 오브젝트 위에 배치 하려면

WorldSpace로 만들어줘야 하는데

이 과정에서는 EventCamera를

![image](https://user-images.githubusercontent.com/43138813/145450434-d50fa524-6fcc-4a8f-8aee-eb032dd59782.png)

이 제페토 플레이어스에 기본으로 탑재되어있는

카메라 프리팹을 사용해야 하는지 여쭤보고 싶습니다.

왜냐하면 현재 버튼이 Canvas의 상태가

Screen Space - Overlay 상태 일때는 버튼이 눌리지만

WorldSpace로 했을때는 눌리지가 않아서요

카메라의 문제인가 싶어서 여쭤봅니다

**Comment by x4cf** [comment]
https://studio.zepeto.me/guides/ui-events
가이드를 참고하시면 좋을것같아요! :)

**Reply by smartbh** [reply]
그냥 캔버스를 만드는게 아니라 빈오브젝트에 스크립트를 넣고 btn을 따로 쓰는 형식인가보네요 좀 특이하네요

**Comment by mhkim0720** [comment]
캔버스이벤트카메라에 제페토카메라를 넣으시면 되요

**Reply by smartbh** [reply]
제페토 카메라 넣고 실행해봤더니 제가 생각하는 오브젝트 위에 뜨질 않고 제페토 카메라의 전체 UI판에서 뜨네요 흠..

**Comment by mhkim0720** [comment]
![image](https://user-images.githubusercontent.com/8485731/145519712-ba83fd96-c7f7-4ff7-bdca-f0fe12becae7.png)
캔버스 월드 좌표로 의자 버튼 구현한 스샷인데 처음 생각하신 대로 구현 하심 될거 같네요

**Reply by smartbh** [reply]
와 엄청 깔끔하고 멋지게 하셨네요; 부럽습니다.
